### PR TITLE
Add note to RawSerial.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Add deprecation note.

### CODE REVIEW

File: docs/api/io/RawSerial.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # RawSerial
 
+<span class="notes">**Note:** Some functions, variables or types have been deprecated. Please see the class reference linked below for details.</span>
```
Comment (hugueskamba) [inline docs/api/io/RawSerial.md:3]: All of it has been deprecated.

Comment (AnotherButler) [reply docs/api/io/RawSerial.md:3]: Is there an API we should direct people to use in its place?

Comment (hugueskamba) [reply docs/api/io/RawSerial.md:3]: Yes, [`UnbufferedSerial`](https://github.com/ARMmbed/mbed-os/blob/master/drivers/UnbufferedSerial.h). Currently working on the docs.

**Comment by AnotherButler** [comment]
Duplicates work of #1241

**Pull request closed without merging**